drivers/flash/spi_nor.c

```diff
@@ -1093,27 +1093,37 @@ static int spi_nor_set_address_mode(const struct device *dev,
 	}
 
 	/* This currently only supports command 0xB7 (Enter 4-Byte
-	 * Address Mode), with or without preceding WREN.
+	 * Address Mode), with or without preceding WREN. Or by setting BIT(7)
+	 * in the Bank Address Register 0x17, when BIT(3) in enter_4byte_addr
+	 * is set.
 	 */
-	if ((enter_4byte_addr & 0x03) == 0) {
+	if ((enter_4byte_addr & 0x0b) == 0) {
 		return -ENOTSUP;
 	}
 
 	acquire_device(dev);
 
+	if ((enter_4byte_addr & 0x08) != 0) {
+		uint8_t sr = BIT(7);
```

Comment:
Comment here on JEDEC 216F description of Enter 4-Byte Addressing and just use hardcoded 0x17 with explanation in comment.